src/draw/grouping.typ

```diff
@@ -400,7 +401,16 @@
 /// - callback (function): A function that accepts the context dictionary and only returns a new one.
 #let set-ctx(callback) = {
   assert(type(callback) == function)
-  return (ctx => (ctx: callback(ctx)),)
+  return (ctx => {
+    let new-ctx = callback(ctx)
+    assert(new-ctx != none, message: "set-ctx must return a context!")
+
+    if new-ctx.transform != ctx.transform {
+      // User supplied matrices can cause rounding issues
+      new-ctx.transform = matrix.round(new-ctx.transform)
+    }
+    (ctx: new-ctx)
+  },)
 }
```

Comment:
In other places, we do rounding anyway. I do not think it is of any use to not round here. If the user really wants a matrix that makes problems, they can still write a custom element. I do not want to get tickets about rounding errors because the user used `set-ctx` instead of `set-transform`.

Reply:
Okay but it should be mentioned in the function docs

Reply:
RfR.